pkg/server/plugin/keymanager/gcpkms/gcpkms_test.go

```diff
@@ -0,0 +1,1478 @@
+							},
+						},
+					},
+				},
+			},
+			waitForDelete: true,
+			destroyTime:   fakeTime,
+			logs: []spiretest.LogEntry{
+				{
+					Level:   logrus.WarnLevel,
+					Message: "CryptoKeyVersion is not enabled, will not be scheduled for destruction",
+					Data: logrus.Fields{
+						cryptoKeyVersionNameTag:  fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName1),
+						cryptoKeyVersionStateTag: kmspb.CryptoKeyVersion_DISABLED.String(),
+					},
+				},
+			},
+		},
+		{
+			name: "error getting CryptoKeyVersion",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P256,
+			},
+			destroyCryptoKeyVersionErr: errors.New("error scheduling CryptoKeyVersion for destruction"),
+			getCryptoKeyVersionErr:     errors.New("error getting CryptoKeyVersion"),
+			fakeCryptoKeys: []fakeCryptoKey{
+				{
+					CryptoKey: &kmspb.CryptoKey{
+						Name:            cryptoKeyName1,
+						VersionTemplate: &kmspb.CryptoKeyVersionTemplate{Algorithm: kmspb.CryptoKeyVersion_EC_SIGN_P256_SHA256},
+					},
+					fakeCryptoKeyVersions: map[string]*fakeCryptoKeyVersion{
+						"1": {
+							publicKey: pubKey,
+							CryptoKeyVersion: &kmspb.CryptoKeyVersion{
+								Algorithm: kmspb.CryptoKeyVersion_EC_SIGN_P256_SHA256,
+								Name:      fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName1),
+							},
+						},
+					},
+				},
+			},
+			waitForDelete: true,
+			destroyTime:   fakeTime,
+			logs: []spiretest.LogEntry{
+				{
+					Level:   logrus.ErrorLevel,
+					Message: "Could not get the CryptoKeyVersion while trying to schedule it for destruction",
+					Data: logrus.Fields{
+						cryptoKeyVersionNameTag: fmt.Sprintf("%s/cryptoKeyVersions/1", cryptoKeyName1),
+						reasonTag:               "error getting CryptoKeyVersion",
+					},
+				},
+			},
+		},
+		{
+			name:       "error getting token info",
+			expectCode: codes.Internal,
+			expectMsg:  "could not get token information: error getting token info",
+			generateKeyReq: &keymanagerv1.GenerateKeyRequest{
+				KeyId:   spireKeyID1,
+				KeyType: keymanagerv1.KeyType_EC_P384,
+			},
+			getTokenInfoErr: errors.New("error getting token info"),
+		},
+	} {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			ts := setupTest(t)
+			ts.fakeKMSClient.destroyTime = fakeTime
+			ts.fakeKMSClient.putFakeCryptoKeys(tt.fakeCryptoKeys)
+			ts.fakeKMSClient.createCryptoKeyErr = tt.createKeyErr
+			ts.fakeKMSClient.getCryptoKeyVersionErr = tt.getCryptoKeyVersionErr
+			ts.fakeKMSClient.getTokeninfoErr = tt.getTokenInfoErr
+			ts.fakeKMSClient.updateCryptoKeyErr = tt.updateCryptoKeyErr
+			ts.fakeKMSClient.destroyCryptoKeyVersionErr = tt.destroyCryptoKeyVersionErr
+			destroySignal := make(chan error)
+			ts.plugin.hooks.scheduleDestroySignal = destroySignal
+
+			configureReq := tt.configureReq
+			if configureReq == nil {
+				configureReq = configureRequestWithDefaults(t)
+			}
+			_, err := ts.plugin.Configure(ctx, configureReq)
```

Comment:
instead of cofnigure it directly you an use 
```
               	coreConfig := catalog.CoreConfig{
  		       TrustDomain: spiffeid.RequireTrustDomainFromString("example.org"),
                }
                km := new(keymanager.V1)
		plugintest.Load(s.T(), , ts.plugin, km
			plugintest.CaptureConfigureError(&err),
			plugintest.CoreConfig(coreConfig),
			plugintest.Configure(configureReq),
		)
                km.GenerateKey()
```
using this you can be sure that your plugin is able to be configured using standard procedures.